parsers/rwcpopmelody_parser.py

```diff
@@ -0,0 +1,197 @@
+    annot.annotation_metadata.annotator = {}
+
+
+def create_jams(smf_file, out_file, metadata):
+    """
+    Creates a JAMS file from an RWC POP smf file (RM-P*.MID).
+    Note: only the notes of the MELODY track are kept!
+    """
+
+    # Load midi file
+    m = midi.read_midifile(smf_file)
+
+    # This will store the relevant MIDI tracks
+    melody = []
+
+    # Track 0 is metadata that we need
+    melody.append(m[0])
+
+    # Collect track text (convert to lower case and remove spaces)
+    track_text = ["".join(e.text.lower().split()) for t in m
+                  for e in t if isinstance(e, midi.TrackNameEvent)]
+
+    assert len(track_text) == len(m)
+
+    # Find the melody track: it should START with "melo", or if that's not
+    # there then "voca" (for vocal).
+    track_text = [t[:4] for t in track_text]
+    if 'melo' in track_text:
+        index = track_text.index('melo')
+    elif 'voca' in track_text:
+        index = track_text.index('voca')
+    else:
+        print("ABORTING TRACK: couldn't find melody for: %s" % smf_file)
+        return 0
+
+    melody.append(m[index])
+
+    # Create new midi pattern with just these tracks
+    m = midi.Pattern(tracks=melody, resolution=m.resolution,
+                     format=m.format, tick_relative=m.tick_relative)
+
+    # Write temporary midi file
+    temp_midi = out_file.replace(".jams", ".temp.mid")
+    midi.write_midifile(temp_midi, m)
+
+    # Load temp midi file using pretty_midi
+    pm = pretty_midi.PrettyMIDI(temp_midi)
+
+    assert len(pm.instruments) == 2
+    assert pm.instruments[0].notes == []
+    assert len(pm.instruments[1].notes) > 0
+
+    # Get the melody notes
+    notes = pm.instruments[1].notes
+
+    # Create jam
+    jam = jams.JAMS()
+
+    # Create annotation
+    midi_ann = jams.Annotation('pitch_midi')
+
+    # Add notes to the annotation
+    for note in notes:
+        midi_ann.append(time=note.start, duration=(note.end-note.start),
+                        value=note.pitch, confidence=1.)
+
+    # Fill annotation metadata
+    fill_annotation_metadata(midi_ann)
+
+    # Add annotation to jam
+    jam.annotations.append(midi_ann)
+
+    # Fill file metadata
+    n_track = int(os.path.basename(smf_file)[4:7]) - 1
+    fill_file_metadata(jam, metadata, n_track)
+
+    # Save JAMS
+    jam.save(out_file)
+
+    # Remove temporary midi file
+    os.remove(temp_midi)
+
+
+def process_folder(smf_dir, out_dir):
+    """Converts the original SMF annotations into the JAMS format (keeping only
+    the notes of the melody track), and saves them in the out_dir folder."""
+
+    # Collect all SMF annotations.
+    smf_files = jams.util.find_with_extension(smf_dir, '.MID', depth=1)
+
+    # Get metadata
+    metadata = pd.read_csv('data/rwcmelodypop/metadata.csv')
+
+    for smf in smf_files:
+
+        jams_file = (
+            os.path.join(out_dir,
+                         os.path.basename(smf).replace('.MID', '.jams')))
```

Comment:
This will do strange things when you have inputs that have `.MID` multiple times.  It would be better to `splitext` and join the desired extension.